### PR TITLE
Types for Button, IconButton, and Field

### PR DESCRIPTION
**Pull request by annatraussnig**
Because I need them for the editor project.

### CODE REVIEW

**Review by ovbm — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -375,10 +375,10 @@ const Styleguide = () => {
...
                         ...require('./components/Typography'),
                         Button: require('./components/Button'),
-                        Field: require('./components/Form/Field.js'),
-                        Dropdown: require('./components/Form/Dropdown'),
-                        VirtualDropdown: require('./components/Form/VirtualDropdown'),
-                        NativeDropdown: require('./components/Form/NativeDropdown'),
+                        Field: require('./components/Form/Field.tsx'),
+                        Dropdown: require('./components/Form/Dropdown.js'),
+                        VirtualDropdown: require('./components/Form/VirtualDropdown.js'),
+                        NativeDropdown: require('./components/Form/NativeDropdown.js'),
```
Comment (tpreusse) [inline src/index.js:381]:
```suggestion
                        Field: require('./components/Form/Field'),
                        Dropdown: require('./components/Form/Dropdown'),
                        VirtualDropdown: require('./components/Form/VirtualDropdown'),
                        NativeDropdown: require('./components/Form/NativeDropdown'),
```

File: src/index.js [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ const Styleguide = () => {
                       title: 'Translate',
                       src: require('./lib/translate.docs.md'),
                       imports: {
-                        Field: require('./components/Form/Field.js'),
+                        Field: require('./components/Form/Field.tsx'),
```
Comment (tpreusse) [inline src/index.js:1011]:
```suggestion
                        Field: require('./components/Form/Field'),
```

File: src/index.js [hunk 3]
```diff
@@ -401,11 +401,11 @@ const Styleguide = () => {
                           }
                         ],
                         VirtualDropdownInternal: {
-                          Items: require('./components/Form/VirtualDropdown')
+                          Items: require('./components/Form/VirtualDropdown.js')
```
Comment (tpreusse) [inline src/index.js:404]:
```suggestion
                          Items: require('./components/Form/VirtualDropdown')
```

File: src/index.js [hunk 4]
```diff
@@ -401,11 +401,11 @@ const Styleguide = () => {
...
                             .Items,
-                          ItemsContainer: require('./components/Form/VirtualDropdown')
+                          ItemsContainer: require('./components/Form/VirtualDropdown.js')
```
Comment (tpreusse) [inline src/index.js:406]:
```suggestion
                          ItemsContainer: require('./components/Form/VirtualDropdown')
```

File: src/index.js [hunk 5]
```diff
@@ -401,11 +401,11 @@ const Styleguide = () => {
...
                             .ItemsContainer,
-                          Inner: require('./components/Form/VirtualDropdown')
+                          Inner: require('./components/Form/VirtualDropdown.js')
```
Comment (tpreusse) [inline src/index.js:408]:
```suggestion
                          Inner: require('./components/Form/VirtualDropdown')
```

File: src/index.js [hunk 6]
```diff
@@ -337,7 +337,7 @@ const Styleguide = () => {
                         Button: require('./components/Button'),
                         Checkbox: require('./components/Form/Checkbox.js'),
                         Radio: require('./components/Form/Radio.js'),
-                        Field: require('./components/Form/Field.js'),
+                        Field: require('./components/Form/Field.tsx'),
```
Comment (tpreusse) [inline src/index.js:340]: Easiest is to just drop the file name ending:
```suggestion
                        Field: require('./components/Form/Field'),
```
(if you feel like it get ride of all of them in the file)

File: src/components/Button/index.tsx [hunk 7]
```diff
@@ -51,7 +51,24 @@ const styles = {
   })
 }
 
-const Button = React.forwardRef(
+const Button = React.forwardRef<
+  HTMLAnchorElement & HTMLButtonElement,
+  {
+    onClick?: MouseEventHandler<HTMLAnchorElement> &
+      MouseEventHandler<HTMLButtonElement>
```
Comment (trm217) [inline src/components/Button/index.tsx:58]:
```suggestion
    onClick?: MouseEventHandler<HTMLAnchorElement & HTMLButtonElement>
```

File: src/components/Form/Field.tsx [hunk 8]
```diff
@@ -112,7 +118,28 @@ const ArrowDown = ({ size, fill, ...props }) => (
...
+    autoComplete?: boolean
+    type?: string
+    label?: string
+    disabled?: boolean
+    error?: string | boolean
+    onInc?: () => void
+    onDec?: () => void
+    icon?: IconType
+    simulate?: string
+    renderInput: React.FC<Record<string, unknown>>
```
Comment (trm217) [inline src/components/Form/Field.tsx:140]: Looks good for now, but I believe we should specify an interface to define what props are available to an input. If we defined the type and export it, in the Frontend we could easily create new inputs against this interface.

File: src/components/IconButton/index.tsx [hunk 9]
```diff
@@ -1,13 +1,34 @@
...
+    target?: string
+    label?: string
+    labelShort?: string
+    title?: string
+    fill?: string
+    fillColorName: string
+    onClick?: MouseEventHandler<HTMLAnchorElement> &
+      MouseEventHandler<HTMLButtonElement>
+    onMouseDown?: MouseEventHandler<HTMLAnchorElement> &
+      MouseEventHandler<HTMLButtonElement>
```
Comment (trm217) [inline src/components/IconButton/index.tsx:25]: Slight clean up
```suggestion
    onClick?: MouseEventHandler<HTMLAnchorElement & HTMLButtonElement>
    onMouseDown?: MouseEventHandler<HTMLAnchorElement & HTMLButtonElement>
```

File: src/components/Tabs/TabButton.tsx [hunk 10]
```diff
@@ -53,40 +53,48 @@ const styles = {
...
-    const [colorScheme] = useColorContext()
+const TabButton = React.forwardRef<
+  HTMLAnchorElement & HTMLButtonElement,
+  {
+    border?: boolean
+    isActive?: boolean
+    text?: string
+    href?: string
+    onClick?: MouseEventHandler<HTMLAnchorElement> &
+      MouseEventHandler<HTMLButtonElement>
```
Comment (trm217) [inline src/components/Tabs/TabButton.tsx:64]:
```suggestion
    onClick?: MouseEventHandler<HTMLAnchorElement & HTMLButtonElement>
```

**Review by trm217 — COMMENTED** [review]
Small cleanup inside on the type definitions. Besides that everything looks good to me.

File: src/components/Form/Field.tsx (hunk 8 above)
Comment (annatraussnig) [reply src/components/Form/Field.tsx:140]: fully agreed. same goes for icon rest props.

**Pull request merged**